### ISSUE TITLE
Not able to configure HTTPS endpoint using self signed certificate.

### ISSUE THREAD

**Issue by lazyav** [dotnet/aspnetcore#53187]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I am not able to configure HTTPS endpoint using self signed certificate. 

Following are the reference document I am following.

https://learn.microsoft.com/en-us/dotnet/core/additional-tools/self-signed-certificates-guide
https://learn.microsoft.com/en-us/aspnet/core/security/docker-https?view=aspnetcore-8.0
https://github.com/dotnet/dotnet-docker/blob/main/samples/run-aspnetcore-https-development.md

### Expected Behavior

Should be able to browse api application locally on https://localhost:7001/weatherforecast

### Steps To Reproduce

1. Create a new Asp.Net Core Web API Project (Visual Studio Community 2022)
    ![image](https://github.com/dotnet/aspnetcore/assets/106724931/1ed40a61-fde9-4aa9-ba33-e70782b68661)

2. Run the following commands in PS
```
dotnet dev-certs https --clean

dotnet dev-certs https -ep $env:USERPROFILE\.aspnet\https\WeatherApi.pfx -p password

dotnet dev-certs https --trust

dotnet user-secrets -p WeatherApi.csproj set "Kestrel:Certificates:Development:Password" "password"  

docker image build -t weatherapi .

docker run --rm -it -p 7001:8081 -e ASPNETCORE_HTTPS_PORTS=7001 -e ASPNETCORE_ENVIRONMENT=Development -v $env:APPDATA\microsoft\UserSecrets\:/root/.microsoft/usersecrets -v $env:USERPROFILE\.aspnet\https:/root/.aspnet/https/ weatherapi

```  

         ![image](https://github.com/dotnet/aspnetcore/assets/106724931/3f9efd7c-d8f8-4b5c-af1e-134f98568f88)

         ![image](https://github.com/dotnet/aspnetcore/assets/106724931/4836d7c3-b93b-46e0-867f-f2e97183edba)

         ![image](https://github.com/dotnet/aspnetcore/assets/106724931/51e3c68c-df6a-47d4-8bef-87bfd2e5873a)

### Exceptions (if any)

```

fail: Microsoft.Extensions.Hosting.Internal.Host[11]
      Hosting failed to start
      System.InvalidOperationException: Unable to configure HTTPS endpoint. No server certificate was specified, and the default developer certificate could not be found or is out of date.
      To generate a developer certificate run 'dotnet dev-certs https'. To trust the certificate (Windows and macOS only) run 'dotnet dev-certs https --trust'.
      For more information on configuring HTTPS see https://go.microsoft.com/fwlink/?linkid=848054.
         at Microsoft.AspNetCore.Hosting.ListenOptionsHttpsExtensions.UseHttps(ListenOptions listenOptions, Action`1 configureOptions)
         at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.AddressesStrategy.BindAsync(AddressBindContext context, CancellationToken cancellationToken)
         at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindAsync(ListenOptions[] listenOptions, AddressBindContext context, Func`2 useHttps, CancellationToken cancellationToken)
         at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.BindAsync(CancellationToken cancellationToken)
         at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.StartAsync[TContext](IHttpApplication`1 application, CancellationToken cancellationToken)
         at Microsoft.AspNetCore.Hosting.GenericWebHostService.StartAsync(CancellationToken cancellationToken)
         at Microsoft.Extensions.Hosting.Internal.Host.<StartAsync>b__15_1(IHostedService service, CancellationToken token)
         at Microsoft.Extensions.Hosting.Internal.Host.ForeachService[T](IEnumerable`1 services, CancellationToken token, Boolean concurrent, Boolean abortOnFirstException, List`1 exceptions, Func`3 operation)
Unhandled exception. System.InvalidOperationException: Unable to configure HTTPS endpoint. No server certificate was specified, and the default developer certificate could not be found or is out of date.
To generate a developer certificate run 'dotnet dev-certs https'. To trust the certificate (Windows and macOS only) run 'dotnet dev-certs https --trust'.
For more information on configuring HTTPS see https://go.microsoft.com/fwlink/?linkid=848054.
   at Microsoft.AspNetCore.Hosting.ListenOptionsHttpsExtensions.UseHttps(ListenOptions listenOptions, Action`1 configureOptions)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.AddressesStrategy.BindAsync(AddressBindContext context, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindAsync(ListenOptions[] listenOptions, AddressBindContext context, Func`2 useHttps, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.BindAsync(CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.StartAsync[TContext](IHttpApplication`1 application, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Hosting.GenericWebHostService.StartAsync(CancellationToken cancellationToken)
   at Microsoft.Extensions.Hosting.Internal.Host.<StartAsync>b__15_1(IHostedService service, CancellationToken token)
   at Microsoft.Extensions.Hosting.Internal.Host.ForeachService[T](IEnumerable`1 services, CancellationToken token, Boolean concurrent, Boolean abortOnFirstException, List`1 exceptions, Func`3 operation)
   at Microsoft.Extensions.Hosting.Internal.Host.StartAsync(CancellationToken cancellationToken)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.Run(IHost host)
   at Program.<Main>$(String[] args) in /src/Program.cs:line 17
```

### .NET Version

8.0.100

### Anything else?

[WeatherApi.zip](https://github.com/dotnet/aspnetcore/files/13848447/WeatherApi.zip)

**Comment by amcasey** [comment]
I'm probably missing something obvious, but I don't see where you're configuring the certificate path for the app.  I'd expect you to need to include something like `-e ASPNETCORE_Kestrel__Certificates__Default__Path=/https/aspnetapp.pfx` as described [here](https://learn.microsoft.com/en-us/aspnet/core/security/docker-https?view=aspnetcore-8.0#windows-using-linux-containers).  (You're using `Kestrel:Certificates:Development:Password` above and the two have to match - i.e. `Kestrel:Certificates:Default:Password`.)

Basically, you need the pfx file to end up in the container (usually via mounting), the path to the pfx file to be available to the app in the container (usually via configuration), and the password to end up in the container (usually via configuration, but there's probably also a way to do it with user secrets).

**Comment by amcasey** [comment]
(If you do this in VS, F5 should just plumb in the correct configuration without requiring any powershell commands, so I'm tacitly assuming you're intentionally doing something more complicated.  Please let me know if that's not the case.)

**Comment by amcasey** [comment]
Here's the command I see when VS starts the container on my behalf
```
docker run -dt -v "C:\Users\amcasey\vsdbg\vs2017u5:/remote_debugger:rw" -v "C:\Users\amcasey\AppData\Roaming\Microsoft\UserSecrets:/root/.microsoft/usersecrets:ro" -v "C:\Users\amcasey\AppData\Roaming\Microsoft\UserSecrets:/home/app/.microsoft/usersecrets:ro" -v "C:\Users\amcasey\AppData\Roaming\ASP.NET\Https:/root/.aspnet/https:ro" -v "C:\Users\amcasey\AppData\Roaming\ASP.NET\Https:/home/app/.aspnet/https:ro" -v "C:\Program Files\Microsoft Visual Studio\2022\IntPreview\MSBuild\Sdks\Microsoft.Docker.Sdk\tools\TokenService.Proxy\linux-x64\net6.0:/TokenService.Proxy:ro" -v "C:\Program Files\Microsoft Visual Studio\2022\IntPreview\MSBuild\Sdks\Microsoft.Docker.Sdk\tools\HotReloadProxy\linux-x64\net6.0:/HotReloadProxy:ro" -v "C:\Program Files\Microsoft Visual Studio\2022\IntPreview\Common7\IDE\CommonExtensions\Microsoft\HotReload:/HotReloadAgent:ro" -v "C:\Users\amcasey\source\repos\WebApplication25\WebApplication25:/app" -v "C:\Users\amcasey\source\repos\WebApplication25:/src/" -v "C:\.tools\.nuget\packages\:/.nuget/fallbackpackages" -v "C:\Program Files (x86)\Microsoft Visual Studio\Shared\NuGetPackages:/.nuget/fallbackpackages2" -e "ASPNETCORE_LOGGING__CONSOLE__DISABLECOLORS=true" -e "ASPNETCORE_ENVIRONMENT=Development" -e "DOTNET_USE_POLLING_FILE_WATCHER=1" -e "NUGET_PACKAGES=/.nuget/fallbackpackages" -e "NUGET_FALLBACK_PACKAGES=/.nuget/fallbackpackages;/.nuget/fallbackpackages2" -P --name WebApplication25 --entrypoint tail webapplication25:dev -f /dev/null 
```

In particular, note that it mounts things to both `/root` and `/home/app`.  I would guess that's what needs to be updated in your commands (and the sample instructions).  Alternatively, you can configure the certificate path explicitly, as I described in my original response.

**Comment by dovewine** [comment]
I encountered the same problem,it seems to be related to the UserSecrets.
This issue occurs when using Docker, rather than in IIS Express or Kestrel.

If this line is commented out, this problem will occur.
![image](https://github.com/dotnet/aspnetcore/assets/38154124/e78cbf5b-65f3-47a9-b335-f5c721780ee6)

This also applies to DockerComponse.
![image](https://github.com/dotnet/aspnetcore/assets/38154124/1e2c8b62-fe4d-44ae-b10f-87698373a589)

**Comment by YunusOzdemirr** [comment]
> I encountered the same problem,it seems to be related to the UserSecrets. This issue occurs when using Docker, rather than in IIS Express or Kestrel.
> 
> If this line is commented out, this problem will occur. ![image](https://github.com/dotnet/aspnetcore/assets/38154124/e78cbf5b-65f3-47a9-b335-f5c721780ee6)
> 
> This also applies to DockerComponse. ![image](https://github.com/dotnet/aspnetcore/assets/38154124/1e2c8b62-fe4d-44ae-b10f-87698373a589)

This is not releated with docker
